controllers/actions.github.com/resourcebuilder.go

```diff
@@ -759,5 +772,9 @@ func mergeLabels(base, overwrite map[string]string) map[string]string {
 		mergedLabels[k] = v
 	}
 
+	// Remove the label to ensure that resources created by this controller are not managed by ArgoCD.
+	// This prevents ArgoCD from pruning and deleting these resources when syncing.
+	delete(mergedLabels, "argocd.argoproj.io/instance")
```

Comment:
I really think we should pass an array with excluded labels (annotations, etc - anything inherited) keys and make `argocd.argoproj.io/instance` one of the defaults.

In `values.yaml`:
```yaml
# ...
    excludedMergedLabels: # TODO: better naming
      - argocd.argoproj.io/instance
      # - ... 
```

Reply:
Hey, this is the direction we are headed to
I'll prepare a PR. We were discussing the approach and would like to hear your opinions on it.
To make a flag on the controller that is named `exclude-label-prefixes`. This will be a list of label prefixes, so it allows you to ignore the entire namespace of labels.
It would be on the controller, so it would be propagated through all scale sets.
Basically, the way it works is go through labels, and if it starts with the prefix, you ignore it.

That way, if you want to ignore a single label, you specify the full name. Otherwise, you can specify something like `argocd.argoproj.io/`, and it would ignore all labels that are related to argo.

Reply:
@nikola-jokic Okay. Thx.

Reply:
@nikola-jokic @kahirokun

To make this PR through, should @kahirokunn change its content, description and title to only the `ownerReferences` feature?

I think this PR is touching on 2 different issues and maybe should be merged and released right now (pointing out the tech debt) to solve those Argo CD issues; or be split.

Reply:
Okay.
Commits other than ownerReferences dropped.

Reply:
done.
https://github.com/actions/actions-runner-controller/pull/3607